### DISCUSSION TITLE
about  Protocol Detection

### DISCUSSION THREAD

**Discussion by eddielishan** [linkerd/linkerd2#8047] [Q&A]
Is it possible that the connecting to a server-speaks-first destination that is outside of the cluster  will detect the automatic skip port without the configuration (- -skip-outbound-ports) protocol.Otherwise each need to configure the more troublesome...Maintenance costs are also relatively high.Is there a better way?thanks

**Comment by mateiidavid** [comment]
Hey @eddielishan,

* `skip-outbound-ports` will bypass the proxy. Your client and database/server will not involve any proxying so all should be well.

* `opaquePorts` will bypass protocol detection and forward everything as TCP. This will only work inside of your cluster. You cannot mark a target as being opaque outside of your cluster.

* `default` with an off cluster server/database, you'll get a 10s delay. Only option if it's _off-cluster_ is to skip. If it's _in cluster_, you can mark as opaque.

Hope that helps :D
